modules/gapi/src/streaming/gstreamer/gstreamer_pipeline_priv.hpp

```diff
@@ -0,0 +1,67 @@
+//
+// Copyright 2021 Intel Corporation.
+//
+// This software and the related documents are Intel copyrighted materials,
+// and your use of them is governed by the express license under which they
+// were provided to you ("License"). Unless the License provides otherwise,
+// you may not use, modify, copy, publish, distribute, disclose or transmit
+// this software or the related documents without Intel's prior written
+// permission.
+//
+// This software and the related documents are provided as is, with no
+// express or implied warranties, other than those that are expressly
+// stated in the License.
+//
+
+#ifndef OPENCV_GAPI_STREAMING_GSTREAMER_GSTREAMER_PIPELINE_PRIV_HPP
+#define OPENCV_GAPI_STREAMING_GSTREAMER_GSTREAMER_PIPELINE_PRIV_HPP
+
+#include <opencv2/gapi/streaming/gstreamer/gstreamer_pipeline.hpp>
+
+#include <string>
+#include <unordered_map>
+
+#ifdef HAVE_GSTREAMER
+
+namespace cv {
+namespace gapi {
+namespace wip {
+
+class GStreamerPipeline::Priv
+{
+public:
+    explicit Priv(const std::string& pipeline);
+
+    IStreamSource::Ptr getStreamingSource(const std::string& appsinkName,
+                                          const GStreamerSource::OutputType outputType);
+
+private:
+    std::shared_ptr<GStreamerPipelineFacade> m_pipeline;
+    std::unordered_map<std::string, bool> m_appsinkNamesToUse;
```

Comment:
whats's about `std::unordered_map<std::string, Ptr<IStreamSource>> ` where  `Ptr<IStreamSource>` is already created source? to avoid breeding extra bool-entiry. Because if user just destroed `IStreamSource::Ptr` in our client  code part we would get `bool == true` and no source actually

UPDATE: or `IStreamSource` straight  to -> `GstreamerSource`